src/dmtcp_coordinator.cpp

```diff
@@ -660,6 +660,20 @@ DmtcpCoordinator::onData(CoordClient *client)
     client->setState(msg.state);
     client->progname(progname);
     client->identity(msg.from);
+    // We need to reset the alarm.  There are at least two scenarios where
+    // this matters:
+    //   (i) Maybe we used 'gdb' for dmtcp_launch, and the alarm signal
+    //       arrived before we exec'ed from the dmtcp_launch to the executable.
+    //       In that case, no checkpoint occurred and
+    //       'DmtcpCoordinator::releaseBarrier()' failed to be called
+    //       and so resetCkptTimer() was not called again.
+    //  (ii) Maybe 'dmtcp_coordintor -i 5' set a checkpoint interval.
+    //       And maybe one job, 'dmtcp_launch --join-coorindator a.out'
+    //       completed.  After that, there were no jobs, and so, as before,
+    //       resetCkptTimer() was not called at the next alarm, since there
+    //       were no jobs.
+    // So, we reset the ckpt timer now, when a new job is init'ed or exec'ed.
+    resetCkptTimer();
```

Comment:
Resetting alarm here could prevent a checkpoint from happening if the computation creates a new process with higher frequency than the alarm frequency. In that case, we would keep resetting alarm on process creation thus never checkpointing.

Perhaps, a better fix would be to "delay" checkpoint until the computation reaches a "checkpointable" state, i.e., all processes have finished init/exec.